packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -2958,57 +3312,57 @@ class ObjectsInRedisClient {
         }
     }
 
-    getObjectsByPatternAsync(pattern, options) {
+    getObjectsByPatternAsync(pattern: string, options: CallOptions): Promise<ioBroker.Object[] | void> {
         return new Promise((resolve, reject) =>
             this.getObjectsByPattern(pattern, options, (err, objs) => (err ? reject(err) : resolve(objs)))
         );
     }
 
-    async _setObject(id, obj, options, callback) {
+    private async _setObject(
+        id: string,
+        obj: ioBroker.SettableObject,
+        options: CallOptions
+    ): ioBroker.SetObjectPromise {
```

Comment:
Similar to getObject, you can impose restrictions on the object by inferring its type from the ID:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/06f38fe1967d69ed164552b55674bf2d5269e7a1/types/iobroker/index.d.ts#L696-L713